ArduCopter/control_guided.cpp

```diff
@@ -156,17 +156,34 @@ void Copter::guided_angle_control_start()
 }
 
 // guided_set_destination - sets guided mode's target destination
-void Copter::guided_set_destination(const Vector3f& destination)
+// Returns true if the fence is enabled and guided waypoint is within the fence
+// else return false if the waypoint is outside the fence
+bool Copter::guided_set_destination(const Vector3f& destination)
 {
+     bool ret = true;
     // ensure we are in position control mode
     if (guided_mode != Guided_WP) {
         guided_pos_control_start();
     }
 
-    wp_nav.set_wp_destination(destination);
+#if AC_FENCE == ENABLED
+    // ensure not above fence altitude and radius if alt fence or circular is enabled
+    float home_destination_distance;
```

Comment:
It would be cleaner to do this calculation inside check_fence_guided() rather then passing it in. Clutters up the caller code with details it doesn't need to worry about. check_fence_guided() could be renamed to check_fence_location() to indicate if a requested location would be outside of the fence (since there is no guided specific logic in the area). I'd look to @rmackay9  or another copter dev for guidance on that, but thats the only thing that jumps out at me.